napari/_qt/qt_main_window.py

```diff
@@ -661,25 +661,7 @@ def add_plugin_dock_widget(
                 wdg = wdg._magic_widget
             return dock_widget, wdg
 
-        # if the signature is looking a for a napari viewer, pass it.
-        kwargs = {}
-        try:
-            sig = inspect.signature(Widget.__init__)
-        except ValueError:
-            pass
-        else:
-            for param in sig.parameters.values():
-                if param.name == 'napari_viewer':
-                    kwargs['napari_viewer'] = self.qt_viewer.viewer
-                    break
-                if param.annotation in ('napari.viewer.Viewer', Viewer):
-                    kwargs[param.name] = self.qt_viewer.viewer
-                    break
-                # cannot look for param.kind == param.VAR_KEYWORD because
-                # QWidget allows **kwargs but errs on unknown keyword arguments
-
-        # instantiate the widget
-        wdg = Widget(**kwargs)
+        wdg = _instantiate_dock_widget(Widget, self.qt_viewer.viewer)
 
         # Add dock widget
         dock_kwargs.pop('name', None)
```

Comment:
it looks like dock_kwargs is uninitialized here when the npe2 path is taken.

Reply:
Yes indeedy, thanks!  This is why I want mypy everywhere! :)

Reply:
fixed